### PR TITLE
cascade layers : linting and a few fixes

### PR DESCRIPTION
**Pull request by romainmenke**
- applied linting
- removed `postcss-tape` from `devDependencies` (and internal package is used in the mono repo)
- ensured that all node types can be moved to the implicit layer.
- fixed anonymous layer handling

### CODE REVIEW

File: plugins/postcss-cascade-layers/src/index.ts [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
 			let layerCount = 0;
-			let layerOrder = {};
+			const layerOrder = {};
 
 			// 1st walkthrough to rename anon layers and store state (no modification of layer styles)
-			root.walkAtRules("layer", (atRule) => {
+			root.walkAtRules('layer', (atRule) => {
 				// give anonymous layers a name
 				if (!atRule.params) {
+					atRule.raws.afterName = ' ';
```
Comment (romainmenke) [inline plugins/postcss-cascade-layers/src/index.ts:14]: Without this it became `@layeranon0`.
Wasn't aware of this myself.

File: plugins/postcss-cascade-layers/src/index.ts [hunk 2]
```diff
@@ -19,11 +20,15 @@ function postcssCascadeLayers(opts) {
 
 				// check for where a layer has nested layers AND styles outside of those layers
 				atRule.each((node) => {
-					if (node.type == "atrule") {
+					if (node.type == 'atrule' && node.name === 'layer') {
 						hasNestedLayers = true;
-					} else if (node.type == "rule") {
+					} else {
```
Comment (romainmenke) [inline plugins/postcss-cascade-layers/src/index.ts:25]: This actually needs much more complex logic, but is ok for now.
Needs to traverse the entire child tree.

File: plugins/postcss-cascade-layers/src/index.ts [hunk 3]
```diff
@@ -19,11 +20,15 @@ function postcssCascadeLayers(opts) {
...
 						hasUnlayeredStyles = true;
 					}
+
+					if (hasNestedLayers && hasUnlayeredStyles) {
+						return false;
```
Comment (romainmenke) [inline plugins/postcss-cascade-layers/src/index.ts:30]: `return false` stops the loop.

File: plugins/postcss-cascade-layers/src/index.ts (hunk 2 above)
Comment (romainmenke) [reply plugins/postcss-cascade-layers/src/index.ts:25]: I can also be totally wrong about this.
I suspect there can be edge cases with `@keyframes`, `@supports`, `@media`, ...
Easier to handle this once it comes together a bit and we can test things in a browser.

**Comment by romainmenke** [comment]
@oluoluoxenfree This is ready for review.

**Review by oluoluoxenfree — APPROVED** [review]

**Comment by oluoluoxenfree** [comment]
@romainmenke looks good to me, thanks so much!
